### ISSUE TITLE
vNext Menu keyboard tabbing behaves differently in SSR

### ISSUE THREAD

**Issue by czearing** [microsoft/fluentui#22174]
### Environment Information

- **Package version(s)**: "@fluentui/react-components": "^9.0.0-rc.6"
- **Browser and OS versions**: Edge, Windows 10

### Describe the issue:

The vNext Menu component tabbing acts differently when server-side rendered. When the trigger has focus and it is opened, the first menu item is not focused. Instead, the trigger still has focus. The Menu receives focus again when the cursor enters it.

When the inline prop is enabled, tabbing acts as expected and focuses on the first menu item.

Example of issue:
![Menu Bug](https://user-images.githubusercontent.com/7192338/159328258-f53ed199-3872-4c8f-ad88-d3640a7fe4c7.gif)

### Please provide a reproduction of the issue in a codepen:

1. Follow the guide in react components for server-side rendering https://github.com/microsoft/fluentui/blob/master/packages/react-components/src/Concepts/SSR.stories.mdx
2. Download the latest version of `react-components`.
3. Import Menu and attempt opening it using the keyboard.

Here is Github repo with the issue already reproduced: https://github.com/czearing/Menu-Focus-Issue-Example

#### Expected behavior:

The first menu item should have focus when the trigger is opened.

Expected behavior example:
![Menu Expected Behavior](https://user-images.githubusercontent.com/7192338/159329029-b95d55d4-9670-4283-a918-1b9d233d0699.gif)

Expected behavior reproduction:
https://codesandbox.io/s/menu-expected-behavior-f4je76

**Comment by ling1726** [comment]
Thx for the repro @czearing, I managed to see that in the SSR version, after the menu is opened, the ref to the popover is still null here https://github.com/microsoft/fluentui/blob/f46eb9ae59d7241f9e403fc6f934d42a65ac1b6b/packages/react-components/react-menu/src/components/Menu/useMenu.tsx#L207

**Comment by ling1726** [comment]
I've identified the problem. It seems that the culprit is our `useIsSSR()` hook: https://github.com/microsoft/fluentui/blob/c5abb9c480ac94d12aa644d68d30773d77f2a159/packages/react-components/react-utilities/src/ssr/SSRContext.tsx#L71

The SSR flag is needs an extra render to be set. This means that when the `Menu` effect is triggered, the `Portal` does not render because it uses this hook https://github.com/microsoft/fluentui/blob/c5abb9c480ac94d12aa644d68d30773d77f2a159/packages/react-components/react-portal/src/components/Portal/usePortal.ts#L23

Therefore the popover actually requires an extra render to become visible. Which only happens after the effect that handles first focus in `Menu`

**Comment by ling1726** [comment]
Fix waiting on microsoft/tabster#166
